Resources/Private/Templates/Backend/Index.html

```diff
@@ -43,12 +43,12 @@ <h3 class="card-title">Page tree including TCA demo records</h3>
                 </div>
                 <div class="card-footer">
                     <button data-href="{f:be.uri(route: 'help_styleguide', parameters: '{action: \'tcaCreate\'}')}" class="btn btn-default{f:if(condition: '{demoExists} = true', then: ' disabled')} t3js-generator-action" id="t3-tca-pagetree-create">
-                        <i class="fa fa-plus fa-fw"></i>
+                        <core:icon identifier="actions-plus" size="small" />
                         <f:translate key="LLL:EXT:styleguide/Resources/Private/Language/locallang.xlf:tcaCreateButton" />
                     </button>
 
                     <button data-href="{f:be.uri(route: 'help_styleguide', parameters: '{action: \'tcaDelete\'}')}" class="btn btn-danger{f:if(condition: '{demoExists} != true', then: ' disabled')}  t3js-generator-action" id="t3-tca-pagetree-delete">
-                        <i class="fa fa-trash fa-fw"></i>
+                        <core:icon identifier="actions-delete" size="small" />
```

Comment:
The change from `<i>` to use the `<core:icon>` ViewHelper changes the structure of the mark that is generated. There is no "<i>" created, just some nested spans and svg thingy.

The styleguide JavaScript to handle the button click events and starting the progress spinner expects a "i" tag .. which is not there anymore. That's the reason why the acceptance tests are failing now.

https://github.com/TYPO3/styleguide/blob/8224011a9a75219415001c53bdf90e9aa3e6da9c/Resources/Public/JavaScript/processing-indicator.js#L31

Actuall markup for one delete button atm is following:

```html
<button
  data-href="/typo3/module/help/styleguide?token=dc8cbdb43b9c006a1df73cd5c09cb13f3530bdd1&amp;action=tcaDelete"
  class="btn btn-danger  t3js-generator-action"
  id="t3-tca-pagetree-delete"
>
  <span 
    class="t3js-icon icon icon-size-small icon-state-default icon-actions-delete" 
    data-identifier="actions-delete"
  >
	<span class="icon-markup">
      <svg class="icon-color">
 <use link:href="/typo3/sysext/core/Resources/Public/Icons/T3Icons/sprites/actions.svg#actions-delete"></use></svg>
	</span>
	
</span>
                        Delete page tree and all related records
                    </button>
```

Same for the create button(s) I guess. So the JavaScript needs to be adjusted.

Note: Also not sure if we need to ensure that in core, of if in core only the cmdline / code installation is used.

@minapok Does this help you to finish this change (the JavaScript adoption) ?

Could be reproduced locally on the pull-request and execution.

Reply:
Thanks for the detailed hint, will try to fix it!